module/apmfiber/middleware.go

```diff
@@ -0,0 +1,175 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build go1.12
+// +build go1.12
+
+package apmfiber // import "go.elastic.co/apm/module/apmfiber"
+
+import (
+	"net/http"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/valyala/fasthttp"
+
+	"go.elastic.co/apm"
+	"go.elastic.co/apm/module/apmfasthttp"
+	"go.elastic.co/apm/module/apmhttp"
+)
+
+// Middleware returns a new Fiber middleware handler for tracing
+// requests and reporting errors.
+//
+// This middleware will recover and report panics, so it can
+// be used instead of default recover middleware.
+//
+// By default, the middleware will use apm.DefaultTracer.
+// Use WithTracer to specify an alternative tracer.
+// Use WithPanicPropagation to disable panic recover.
+func Middleware(o ...Option) fiber.Handler {
+	m := &middleware{
+		tracer:           apm.DefaultTracer,
+		requestIgnorer:   apmfasthttp.NewDynamicServerRequestIgnorer(apm.DefaultTracer),
+		panicPropagation: false,
+	}
+
+	for _, o := range o {
+		o(m)
+	}
+
+	return m.handle
+}
+
+type middleware struct {
+	tracer           *apm.Tracer
+	requestIgnorer   apmfasthttp.RequestIgnorerFunc
+	panicPropagation bool
+}
+
+func (m *middleware) handle(c *fiber.Ctx) error {
+	reqCtx := c.Context()
+	if !m.tracer.Recording() || m.requestIgnorer(reqCtx) {
+		return c.Next()
+	}
+
+	name := string(reqCtx.Method()) + " " + c.Path()
+	tx, body, err := apmfasthttp.StartTransactionWithBody(m.tracer, name, reqCtx)
+	if err != nil {
+		reqCtx.Error(err.Error(), fasthttp.StatusInternalServerError)
+
+		return err
+	}
+
+	defer func() {
+		resp := c.Response()
+		path := c.Route().Path
+		if path == "/" && resp.StatusCode() == http.StatusNotFound {
+			tx.Name = string(reqCtx.Method()) + " unknown route"
+		} else {
+			// Workaround for set tx.Name as template path, not absolute
+			tx.Name = string(reqCtx.Method()) + " " + path
+		}
+
+		if v := recover(); v != nil {
+			if m.panicPropagation {
+				panic(v)
```

Comment:
```suggestion
				defer panic(v)
```

I realise this is already inside a defer, but it should still be deferred to ensure the error is captured below.

Reply:
that's my fault :_), thx for attentiveness and patience